File: 217. Contains Duplicate/217.js
// 217. Contains Duplicate solution in JavaScript
// https://leetcode.com/problems/contains-duplicate/description/

/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var containsDuplicate = function(nums) {
    let set = new Set();

    for (num of nums) {
        if (set.has(num)) return true;

        set.add(num);
    }

    return false;
};

File: 217. Contains Duplicate/217.go
// 217. Contains Duplicate solution in GoLang
// https://leetcode.com/problems/contains-duplicate/description/

func containsDuplicate(nums []int) bool {
    num_set := make(map[int]bool)

    for _, num := range nums {
        if num_set[num] == true {
            return true
        }

        num_set[num] = true
    }

    return false
}